privacyidea/lib/token.py

```diff
@@ -3006,18 +3010,34 @@ def import_tokens(tokens: list[dict], update_existing_tokens: bool = True) -> To
         serial = token_info_dict.get("serial")
         try:
             existing_token = get_one_token(serial=serial, silent_fail=True)
-            if not existing_token:
-                token_type = token_info_dict.get("type")
-                db_token = Token(serial, tokentype=token_type.lower())
-                token = create_tokenclass_object(db_token)
+            # We check if there is no existing token or if we want to update existing tokens
+            if not existing_token or update_existing_tokens:
+
+                # We create a new token, if there is no existing token
+                if not existing_token:
+                    token_type = token_info_dict.get("type")
+                    db_token = Token(serial, tokentype=token_type.lower())
+                    db_token.save()
+                    token = create_tokenclass_object(db_token)
+                # We use the existing token and update it
+                else:
+                    token = existing_token
+
+                # Assign the user, if wanted and if there is a user in the token info dict
+                if assign_to_user and token_info_dict.get("user"):
+                    owner = User(login=token_info_dict.get("user").get("login"),
+                                 resolver=token_info_dict.get("user").get("resolver"),
+                                 realm=token_info_dict.get("user").get("realm"),
+                                 uid=token_info_dict.get("user").get("uid"))
+                    token.add_user(owner, override=True)
```

Comment:
Same here, this can throw an exception when the token is assigned to another user.